coriolis/tests/api/v1/test_minion_pools.py

```diff
@@ -0,0 +1,196 @@
+            )
+
+    @ddt.file_data('data/minion_pool_numeric_values.yml')
+    def test__check_pool_numeric_values(
+        self,
+        config,
+        exception_raised
+    ):
+        minimum_minions = config.get("minimum_minions", None)
+        maximum_minions = config.get("maximum_minions", None)
+        minion_max_idle_time = config.get("minion_max_idle_time", None)
+        if exception_raised:
+            self.assertRaisesRegex(
+                Exception,
+                exception_raised,
+                self.minion_pools._check_pool_numeric_values,
+                minimum_minions,
+                maximum_minions,
+                minion_max_idle_time
+            )
+
+    @mock.patch.object(minion_pools.MinionPoolController,
+                       '_check_pool_retention_strategy')
+    @mock.patch.object(minion_pools.MinionPoolController,
+                       '_check_pool_numeric_values')
+    @ddt.file_data('data/minion_validate_create_body.yml')
+    def test__validate_create_body(
+        self,
+        mock__check_pool_numeric_values,
+        mock__check_pool_retention_strategy,
+        config,
+        exception_raised,
+    ):
+        expected_validation_api_method = config.get(
+            "expected_validation_api_method", None)
+        ctxt = {}
+        body = config["body"]
+        body["minion_pool"]["os_type"] = getattr(
+            constants, body["minion_pool"]["os_type"], None)
+        body["minion_pool"]["platform"] = getattr(
+            constants, body["minion_pool"]["platform"], None)
+        body["minion_pool"]["minion_retention_strategy"] = getattr(
+            constants, body["minion_pool"]["minion_retention_strategy"], None)
+
+        if exception_raised:
+            self.assertRaisesRegex(
+                Exception,
+                exception_raised,
+                testutils.get_wrapped_function(
+                    self.minion_pools._validate_create_body),
+                self.minion_pools,
+                ctxt,
+                body
+            )
+        else:
+            with mock.patch.object(
+                endpoints_api.API,
+                expected_validation_api_method) as mock_validation_api_method:
+
+                testutils.get_wrapped_function(
+                    self.minion_pools._validate_create_body)(
+                        self.minion_pools,
+                        ctxt,
+                        body,
+                )
+
+                mock_validation_api_method.assert_called_once_with(
+                    ctxt,
+                    body["minion_pool"]["endpoint_id"],
+                    body["minion_pool"]["environment_options"]
+                )
+                mock__check_pool_numeric_values.assert_called_once_with(
+                    1, 1, 1)
+                mock__check_pool_retention_strategy.assert_called_once_with(
+                    body["minion_pool"]["minion_retention_strategy"]
+                )
```

Comment:
A bit of a long shot, but since you can control this from the yaml file, can you please have all the 11 values passed as `expected_result`? You can pass them as a list:
```yml
- config:
    body:
      minion_pool:
        name: "mock_name"
        endpoint_id: "mock_endpoint_id"
        os_type: "OS_TYPE_LINUX"
        platform: "PROVIDER_PLATFORM_SOURCE"
        environment_options: "mock_environment_options"
        notes: "mock_notes"
        minion_retention_strategy: "MINION_POOL_MACHINE_RETENTION_STRATEGY_DELETE"
    expected_validation_api_method: "validate_endpoint_source_minion_pool_options"
  exception_raised: False
  expected_result:
    - mock_name
    - mock_endpoint_id
    - source
    - linux
    - mock_environment_options
    - 1 # you could also set custom values to these inside config['body']['minion_pool']
    - 1
    - 1
    - delete
    - mock_notes
    - false
```

Then inside your method, you can do the following check:
```python
                result = testutils.get_wrapped_function(
                    self.minion_pools._validate_create_body)(
                        self.minion_pools,
                        ctxt,
                        body,
                )
                self.assertEqual(result, tuple(expected_result))
```